### DISCUSSION TITLE
What can i do for my future？Please help me。

### DISCUSSION THREAD

**Discussion by GtiHuberZZ** [community/community#142393] [Programming Help]
I am a junior undergraduate student in China, majoring in Computer Science and Technology. 

I am unsure about how to plan my future. I have considered becoming a programmer, but I've heard many stories about programmers being forced to work late into the night without overtime pay, having no personal life, and facing layoffs after the age of 35, even though their salaries are not particularly high.

 I've also thought about becoming a civil servant, which might offer more stability, but it's quite challenging to pass the exams. I have no idea which path to take in the future. Please help me; 

I am willing to pay for guidance.

**Comment by slowboii** [comment]
> I am a junior undergraduate student in China, majoring in Computer Science and Technology. 
> 
> I am unsure about how to plan my future. I have considered becoming a programmer, but I've heard many stories about programmers being forced to work late into the night without overtime pay, having no personal life, and facing layoffs after the age of 35, even though their salaries are not particularly high.
> 
>  I've also thought about becoming a civil servant, which might offer more stability, but it's quite challenging to pass the exams. I have no idea which path to take in the future. Please help me; 
> 
> I am willing to pay for guidance.
> 

Have you considered going where the grass is greaner, and the air is not pure poison? Joking ofcource, yeah sorry for not being helpful.

**Comment by Joaquintinsin** [comment]
Hello! I'm a programmer myself, studying at the National University of Río Cuarto in Córdoba, Argentina. What I've learned through my journey is that, despite economic and social challenges, pursuing a career to become a professional should ultimately be something you truly enjoy.

For me, programming isn’t something I take lightly; I genuinely like what I do. I found a specific area of programming that resonates with me, and that's what keeps me motivated. Even though there are tough stories about job instability or long hours, I believe that when you’re passionate about your work, it helps you stay focused on growing and finding the right opportunities for you.

My advice is to think about what aspect of your studies or career truly interests you. When you enjoy what you're doing, it’s easier to overcome the challenges.

**Comment by EliasDeHondt** [comment]
You’re at a pivotal decision point, and it’s normal to feel uncertain. Consider exploring the tech industry further, as there are many sectors with more flexible work hours than traditional programming jobs. Freelancing might give you the independence you’re looking for without fully committing to a permanent role. Tech-related government positions also offer stability and work-life balance, especially if you’re open to taking the necessary exams. Teaching and consulting roles in tech are additional options where your knowledge can be valuable beyond constant programming. Further studies could also help you build expertise, leading to senior positions or research opportunities. Connecting with a mentor or career coach could provide specific insights and help you create a personalized plan for navigating these choices.

**Comment by LuckyMan564** [comment]
What can I do for my future?
This is often questions to everyone. But this is also a new chance to overcome another challenge. If you want to do easy thing, easy job for you, you can never overcome any challenge. But imagine that you passed one challenge before, the new challenge is no problem.
I think above question is same. First, do anything with you like best, so know yourself who I am and what I can do for me and my family?
At the end of this comment, I am a developer and work full time. But I am very proud about working hard because I have to do this for me and my family.
Thanks for your time.

**Comment by MasterInQuestion** [comment]
天下乌鸦一般黑...
+    Whole world all ravens all same black... whatsoever.

    Investing traditional businesses with informatics expertise should work, likely exceptionally well.
    Ability to sense the trend before its coming; or forge the trend:
    Always positive gain for whatsoever business.
    That bank robbers cannot compete.

    But life, shouldn't really be all about businesses.
    岁月无穷极: 会合安可知?
+    Months years without ever ending: Gaining not how may tell?
    安能苦一身: 与世同举措?
+    How could bitter one self: With world together same practice?

    人生得意须尽欢: 莫使金樽空对月.
+    Life at high should definitely enjoy: Let not Golden Cup empty toward Moon.
    人生万事不如意: 明朝散发弄扁舟.
+    Life of all things unbefitting will: Tomorrow loosen hairs making lone boat.
\
\
    Whatever, maybe it's some "何不食肉糜?" ("Why not meal meat soup?") of my view.
    I don't run alike businesses.
    Nor do I have to.

    "终岁无公事: 随岁有贡钱."
+    Whole year without Official affairs: By years with tribute money.
